warehouse/core/src/main/java/datawave/query/data/UUIDTransform.java

```diff
@@ -0,0 +1,43 @@
+package datawave.query.data;
+
+import java.util.regex.Pattern;
+
+/**
+ * Transform UUID rule, defines a pattern and replacement to be applied to a UUID value
+ */
+public class UUIDTransform {
```

Comment:
It seems to me that we could create a dedicated transform for what you want.  Using a regex and replacement mechanism seems expensive.